graphstorm-processing/graphstorm_processing/distributed_executor.py

```diff
@@ -0,0 +1,342 @@
+    """Arguments for the graph processing task"""
+
+    input_prefix: str
+    output_prefix: str
+    config_filename: str
+    num_output_files: int
+    add_reverse_edges: bool
+    log_level: str
+
+
+class DistributedExecutor:
+    """
+    Used as the entry point for distributed execution.
+
+    Parameters
+    ----------
+    executor_config : ExecutorConfig
+        Executor configuration object.
+    """
+
+    def __init__(
+        self,
+        executor_config: ExecutorConfig,
+    ):
+        self.local_config_path = executor_config.local_config_path
+        self.local_output_path = executor_config.local_output_path
+        self.input_prefix = executor_config.input_prefix
+        self.output_prefix = executor_config.output_prefix
+        self.num_output_files = executor_config.num_output_files
+        self.config_filename = executor_config.config_filename
+        self.filesystem_type = executor_config.filesystem_type
+        self.sm_execution = executor_config.sm_execution
+        self.add_reverse_edges = executor_config.add_reverse_edges
+
+        graph_conf = os.path.join(self.local_config_path, self.config_filename)
+        with open(graph_conf, "r", encoding="utf-8") as f:
+            dataset_config_dict: Dict[str, Any] = json.load(f)
+
+        if "version" in dataset_config_dict:
+            self.config_version = dataset_config_dict["version"]
+            self.graph_config_dict = dataset_config_dict["graph"]
+        else:
+            # TODO: Change once GConstruct adds a version to their config spec
+            self.config_version = "gconstruct"
+            converter = GConstructConfigConverter()
+            self.graph_config_dict = converter.convert_to_gsprocessing(dataset_config_dict)["graph"]
+
+        # Create the Spark session for execution
+        self.spark = spark_utils.create_spark_session(self.sm_execution, self.filesystem_type)
+
+    def run(self) -> None:
+        """
+        Executes the Spark processing job.
+        """
+        print("Performing data processing with PySpark...")
+        data_configs = create_config_objects(self.graph_config_dict)
+
+        t0 = time.time()
+        print("Constructing DGLGraph for Heterogeneous Graph")
+        # Prefer explicit arguments for clarity
+        loader = DistHeterogeneousGraphLoader(
+            spark=self.spark,
+            local_input_path=self.local_config_path,
+            local_output_path=self.local_output_path,
+            data_configs=data_configs,
+            input_prefix=self.input_prefix,
+            output_prefix=self.output_prefix,
+            num_output_files=self.num_output_files,
+            add_reverse_edges=self.add_reverse_edges,
+            enable_assertions=False,
+        )
+        loader.load()
+
+        # This is used to upload the output JSON files to S3 on local runs,
+        # since we can't rely on SageMaker to do it
+        if not self.sm_execution and self.filesystem_type == "s3":
```

Comment:
Can we check whether the output_prefix is valid before call loader.load()?

Reply:
Maybe we also need to do the same thing for training and inference.

Reply:
> output_prefix is valid

You mean ensure that the path exists and we have write access to it? I think that should be doable, we'd just need to do that both for S3 and local.

Reply:
Yeah, you are right.